From: =?utf-8?q?Paul_Li=C3=A9tar?= <paul@lietar.net>
Date: Mon, 18 Aug 2014 16:02:16 +0200
Subject: Backend: Fix system locale detection.

The locale detection would crash if no OCR tools are available, or if
system locale cannot be determined (i.e. LANG env var is not set)
---
 src/paperwork/backend/config.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/paperwork/backend/config.py b/src/paperwork/backend/config.py
index 59f798a..ac073cc 100644
--- a/src/paperwork/backend/config.py
+++ b/src/paperwork/backend/config.py
@@ -156,11 +156,14 @@ class PaperworkConfig(object):
         # the best OCR language
 
         ocr_tools = pyocr.pyocr.get_available_tools()
-        if (len(ocr_tools) < 0):
+        if len(ocr_tools) == 0:
             return self.DEFAULT_OCR_LANG
         ocr_langs = ocr_tools[0].get_available_languages()
 
         default_locale_long = locale.getdefaultlocale()[0]
+        if default_locale_long is None:
+            return self.DEFAULT_OCR_LANG
+
         # Usually something like "fr_FR" --> we just need the first part
         default_locale = default_locale_long.split("_")[0]
         try:
@@ -171,7 +174,7 @@ class PaperworkConfig(object):
         except Exception, exc:
             logger.error("Warning: Failed to figure out system language"
                    " (locale is [%s]). Will default to %s"
-                   % (default_locale_long, default_locale_long))
+                   % (default_locale_long, self.DEFAULT_OCR_LANG))
             logger.error('Exception was: %s' % exc)
         return self.DEFAULT_OCR_LANG
 
